src/IssueViz/Editor/LanguageDetection/SonarLanguageRecognizer.cs

```diff
@@ -26,32 +26,26 @@
 using Microsoft.VisualStudio.Utilities;
 using SonarLint.VisualStudio.Core.Analysis;
 
-namespace SonarLint.VisualStudio.Integration.Vsix
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.LanguageDetection
 {
+    public interface ISonarLanguageRecognizer
+    {
+        IEnumerable<AnalysisLanguage> Detect(string filePath, IContentType bufferContentType);
+    }
+
     [Export(typeof(ISonarLanguageRecognizer))]
-    internal class SonarLanguageRecognizer : ISonarLanguageRecognizer
+    public class SonarLanguageRecognizer : ISonarLanguageRecognizer
```

Comment:
Does this class need to be public? MEF will create internal classes, and the any imports will be using the public interface.